rdt-app/src/main/java/io/ona/rdt_app/widget/RDTCountdownTimerFactory.java

```diff
@@ -11,20 +12,30 @@
 
 import java.util.List;
 
+import io.ona.rdt_app.R;
+
 public class RDTCountdownTimerFactory extends CountDownTimerFactory {
 
+    private static final String COUNTDOWN_TIMER_KEY = "countdown_timer";
+    private static final String COUNTDOWN_TIMER_RESULT_READY_KEY = "countdown_timer_results_ready";
     private JsonFormFragment formFragment;
+    private JSONObject stepObject;
 
     @Override
     public List<View> getViewsFromJson(String stepName, Context context, JsonFormFragment formFragment, JSONObject jsonObject, CommonListener listener, boolean popup) throws Exception {
         this.formFragment = formFragment;
+        this.stepObject = jsonObject;
         return super.getViewsFromJson(stepName, context, formFragment, jsonObject, listener, popup);
     }
 
     @Override
     public void onCountdownFinish(Context context) {
         super.onCountdownFinish(context);
-        // Go to next
-        formFragment.next();
+        if (stepObject.optString(JsonFormConstants.KEY).equals(COUNTDOWN_TIMER_KEY)) {
```

Comment:
This can just be the default behaviour under the else block.

Then you can check for the specific `COUNTDOWN_TIMER_RESULT_READY_KEY` and provide custom logic.